### ISSUE TITLE
oras should be able to tag the same digest with different tags

### ISSUE THREAD

**Issue by milosgajdos** [oras-project/oras#529]
We've been experimenting with `oras` for a bit while and came across a rather _unexpected_ behaviour. Specifically, `oras` cli does not seem to be able to tag the same [manifest] `digest` with different tags. 

**Example scenario:** I want to push some content to a registry and tag it with a tag. Then I want to tag the same content with a _different_ tag. The latter does not happen. My _suspicion_ is on the RE-tag, `oras` does `HEAD` request on manifest digest but not on the `tag` -- it finds the digest exists and never follows up with the new tag. 

Is there any specific reason for this behaviour? I can't think of any.

How to reproduce this:

Push some content to a registry:
```shell
$ oras milosgajdos/foo:1.2.3 foo.txt:application/vnd.unknown.layer.v1+txt
Uploading 63c2b3576834 foo.txt
Pushed milosgajdos/foo:1.2.3
Digest: sha256:6f2ee3eaa3f6fc53ce908fa95fe3e0efa49687b37872e27ad88bba8fd74fe3c1
```

Push a new tag for the same content:
```shell
$ oras milosgajdos/foo:4.5.6 foo.txt:application/vnd.unknown.layer.v1+txt
Exists    6f2ee3eaa3f6
Pushed milosgajdos/foo:4.5.6
Digest: sha256:6f2ee3eaa3f6fc53ce908fa95fe3e0efa49687b37872e27ad88bba8fd74fe3c1
```

The worst thing is, the `oras` on re-TAG (i.e. push with different tag) prints output that seems to suggest that the new tag has been pushed:

```shell
Pushed milosgajdos/foo:4.5.6
```

But that doesn't seem to be the case.

**Comment by yizha1** [comment]
@milosgajdos  Could you verify whether this issue is fixed in ORAS CLI 0.14.0? Thanks.

**Comment by milosgajdos** [comment]
This seems to be fixed. Thanks!
